test/debug_info/deferred_len.f90

```diff
@@ -0,0 +1,17 @@
+!RUN: %flang -g -S -emit-llvm %s -o - | FileCheck %s
+
+!CHECK: !DILocalVariable(name: "defl_string"{{.*}}, type: ![[DERIVEDSTRING:[0-9]+]]
+!CHECK: ![[DERIVEDSTRING]] = !DIDerivedType(tag: DW_TAG_pointer_type, baseType: ![[STRING:[0-9]+]]
+!CHECK: ![[STRING]] = !DIStringType(name: "character(*)", stringLength: !{{[0-9]+}}
+
+program deferredlength
+
+        character(len=100) :: buffer
+        character(len=:), allocatable :: defl_string
+
+        read(*,*) buffer
+        defl_string = trim(buffer)
+        print *,defl_string
+
+end program deferredlength
+
```

Comment:
Please remove overhead free line at the end of file. Single EOL at EOF is sufficient. Too many EOLs or no EOL at all is not.

Reply:
@pawosm-arm, thanks for the review and approval. i will address this nit comment.

Reply:
addressed both the NOT comments.